File: examples/Nodejs/fill_form.js
#!/usr/bin/env node

var scrapy = require('scrapystreaming');
var jsonfile = require('jsonfile');

scrapy.createSpider('form', [], function(response) {});

var fromResponseRequest = {
    formdata: {
        custname: 'Sample',
        custemail: 'email@example.com'
    }
};

scrapy.sendFromResponseRequest('http://httpbin.org/forms/post', function(response) {
    jsonfile.writeFile('outputs/fill_form.json', response.body);
    scrapy.closeSpider();
}, fromResponseRequest);

scrapy.runSpider();

File: examples/R/fill_form.R
#!/usr/bin/env Rscript
suppressMessages(library(scrapystreaming))

save_data <- function(response) {
    f <- file("outputs/fill_form.json")
    # serialize the extracted data and close the spider
    write(response$body, f)
    close(f);
    close_spider()
}

create_spider("form", character(0))
from_response_request <- data.frame(formdata = character(1))
from_response_request$formdata <- data.frame(custname = "Sample", custemail = "email@example.com")

send_from_response_request("http://httpbin.org/forms/post", save_data, from_response_request)

run_spider()
